### PR TITLE
Merchants index page

### LINKED ISSUE #34: User Story 39, Merchant Index Page
```
As a visitor
When I visit the merchant's index page at "/merchants"
I see all merchants in the system who are active
Next to each merchant's name I see their city and state
I also see the date they registered
```

DO THESE STORIES FIRST:
- User Story 2, Visitor Navigation

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #35

child of #82

depends on #79

### PR DESCRIPTION
**Pull request by renecasco**
- [] Check this if the PR has been approved by the team to be a quick patch and the rest of this form will not be filled out

# Test Coverage Percentage (Model/Feature):
- rspec - 100%
- rspec spec/models - 100%

# What functionality does this accomplish?

Merchants Index page
# What did you struggle on to complete?

# Implements/Fixes:
* description
closes #34 

# Testing Changes
- [x] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

# Please include an emoji of how you feel about this branch:
👍

### CODE REVIEW

File: app/controllers/merchants_controller.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 class MerchantsController < ApplicationController
 
   def index
+    @merchants = User.where(role: 'merchant')
```
Comment (milevy1) [inline app/controllers/merchants_controller.rb]: Change to: `@merchants = User.where(role: 'merchant', enabled: true)`

File: spec/features/merchants/merchants_index_page_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      within "#merchant-#{@merchant_2.id}" do
+        expect(page).to have_link(@merchant_2.name, href: merchant_path(@merchant_2))
+        expect(page).to have_content(@merchant_2.city)
+        expect(page).to have_content(@merchant_2.state)
+        expect(page).to have_content(@merchant_2.created_at)
+      end
+
+      expect(page).to_not have_link(@merchant_3.name, href: user_path(@merchant_3))
+      expect(page).to have_content(@merchant_3.created_at)
```
Comment (milevy1) [inline spec/features/merchants/merchants_index_page_spec.rb]: I think you meant this line to be a `.to_not`
Its failing on my machine locally though because its the same created_at as merchant_1 and 2.  Maybe just test to not have the name, city, state instead.

File: app/controllers/merchants_controller.rb (hunk 1 above)
Comment (renecasco) [reply app/controllers/merchants_controller.rb]: good catch, done!

File: spec/features/merchants/merchants_index_page_spec.rb (hunk 2 above)
Comment (renecasco) [reply spec/features/merchants/merchants_index_page_spec.rb]: Those can all be repeated just like the timestamp. So I just changed it to_not have the user id in the css, which is a unique attribute.

**Comment by renecasco** [comment]
conflict resolved
